pkg/storage/ingest/config.go

```diff
@@ -178,6 +181,8 @@ func (cfg *KafkaConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet)
 
 	f.DurationVar(&cfg.WaitStrongReadConsistencyTimeout, prefix+"wait-strong-read-consistency-timeout", 20*time.Second, "The maximum allowed for a read requests processed by an ingester to wait until strong read consistency is enforced. 0 to disable the timeout.")
 
+	f.IntVar(&cfg.ProducerSupportedRecordVersion, prefix+"producer-supported-record-version", 0, "The record version that this producer prefers.")
```

Comment:
[nit] The description says "The record version that this producer prefers" but is it really a prefer? I think it's a "must use", no?

Reply:
Yes, I wrote these toggles before the code and kept the wording a bit intentionally ambiguous. It needs refining.

Reply:
- Changed to `...this producer sends`
- Removed `supported` from the flag name. Now it's just `producer-record-version`